org.eclipse.winery.repository.ui/src/app/instance/sharedComponents/topologyTemplate/topologyTemplate.component.ts

```diff
@@ -16,45 +16,52 @@ import { InstanceService } from '../../instance.service';
 import { backendBaseURL, oldTopologyModelerURL, topologyModelerURL } from '../../../configuration';
 import { DomSanitizer, SafeResourceUrl } from '@angular/platform-browser';
 import { WineryVersion } from '../../../model/wineryVersion';
-import { ModalDirective } from 'ngx-bootstrap';
+import { BsModalRef, BsModalService, ModalDirective } from 'ngx-bootstrap';
 import { ActivatedRoute } from '@angular/router';
 import { ToscaTypes } from '../../../model/enums';
 
 @Component({
-    templateUrl: 'topologyTemplate.component.html'
+    templateUrl: 'topologyTemplate.component.html',
 })
 export class TopologyTemplateComponent implements OnInit {
 
+    readonly uiURL = encodeURIComponent(window.location.origin + window.location.pathname + '#/');
+
     loading = true;
     templateUrl: SafeResourceUrl;
     editorUrl: string;
     oldEditorUrl: string;
+    refinementAvailable = false;
 
     selectedVersion: WineryVersion;
 
     @ViewChild('compareToModal') compareToModal: ModalDirective;
+    compareToModalRef: BsModalRef;
+
+    refinementIsRunning: boolean;
```

Comment:
enum refinementstatus may improve readability :)

Reply:
I tried, but it just did not work... I had errors when using the enum in the template by using the way of creating a special variable for the enum to reference in the template.